### PR TITLE
drivers: Rename cc310 to cc3xx

### PR DESCRIPTION
**Pull request by joerchan**
Rename the hw_cc310 to hw_cc3xx to reflect its dual cc310 and cc312
support as well aligning it with the option HW_CC3XX and the rest of
the cryptocell naming.

Update documentation to match existing entropy_cc310 to entropy_cc3xx
renaming.

Remove documentation describing old behavior of the entropy_cc3xx
driver. This no longer calls the psa_generate_random with TF-M enabled.
Instead there is a separate PSA entropy driver.

### CODE REVIEW

**Review by SebastianBoe — APPROVED** [review]

File: doc/nrf/drivers/hw_cc310.rst [hunk 1]
```diff
@@ -1,25 +1,26 @@
 .. _lib_hw_cc310:
+.. _lib_hw_cc3xx:
 
-CC310 hardware driver
+CC3XX hardware driver
```
Comment (mia-ko) [inline doc/nrf/drivers/hw_cc310.rst:4]: Should the file name be renamed as well?

**Review by mia-ko — CHANGES_REQUESTED** [review]
We should probably mention this in a change log entry :)

**Comment by joerchan** [comment]
> We should probably mention this in a change log entry :)

I don' think there is any point. Nothing has changed for the user. The Kconfig options were all named cc3xx already.

File: doc/nrf/drivers/hw_cc310.rst (hunk 1 above)
Comment (joerchan) [reply doc/nrf/drivers/hw_cc310.rst:4]: Yes, thank you.

File: doc/nrf/drivers/entropy_cc3xx.rst [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+.. _lib_entropy_cc3xx:
+
+CC3XX entropy driver
+####################
+
+.. contents::
+   :local:
+   :depth: 2
+
+You can use the CC310 entropy driver (*entropy_cc3xx*) to generate random data using the Arm CryptoCell CC31XX hardware available on the following Nordic SoCs/SiPs:
```
[the remark this answers is not in the document]
Comment (joerchan) [reply doc/nrf/drivers/entropy_cc3xx.rst]: Typo, should be cc3xx

**Review by mia-ko — APPROVED** [review]

**Pull request merged**